### ISSUE TITLE
feature: create simple search (unified) collection

### ISSUE THREAD

**Issue by gregorpirgie** [acdh-oeaw/viecpro-backend-devops#13]
create a typesense collection that contains all searchable entities present in the database

**Comment by sennierer** [comment]
just because we had a look at this today: typesense has a federated search feature that allows to search multiple collections at the same time -> so as far as I can tell, no need for a unified collection

**Comment by gregorpirgie** [comment]
@sennierer I am a little bit surprised by this message. First: @stefanprobst, @kaggl  and me have been discussing the pros and cons of a unified collection vs. a federated search implementation a while ago already. This you couldn't know; but I re-iterrared the arguments for the unified collection  – which was also Stefan's preference -  and the implementation implications of a federated search  in the "viecpro board of director's" meeting weeks ago, and we decided on this approach towards the end of the meeting.

I am always open to re-evaluate the decision, but given that this is a lower priority feature atm  (in accordance with the view of the viecpro editoral team) I would postpone this until more urgent issues have been resolved. 

And an important sidenote: This is not the first time that I see that several people had to re-discover features or implications of a tool that we use across projects; and I see a deeper issue there that we lack a place where we can reliably deposit such information or a structure or space where we can share such information regularly so that people actually can find it.

That was a main line in my ongoing argumentation for a different place to deposit such information than in github issues (where people might stumble across them, but will most likely never find them if they actively search). When I last brought that up in the apis-jfx (and I did that several times), nobody saw the urgency to change something so that we don't have to do the same work over and over again.

**Comment by sennierer** [comment]
I didnt recall that we had already discussed this. If we already decided against federated search I am fine with that, just wanted to mention it here as we looked at it yesterday for another topic.
And regarding the place to deposit information: even if we had (yet another) place to put information in, chances are high I would have missed that information too.
Anyhow, now its where it should be, with the issue for implementing the unified collection. I also agree that the unified search is not high priority.

**Comment by gregorpirgie** [comment]
Yeah, sure, I will let you know when I start working on this in case something new might have come up. And don't get me wrong – thank you for pointing this out; of course the feature you hinted to might have slipped my attention; this can always happen. 

Regarding the documentation I still (respectfully) disagree, but thats neither the topic now, nor a pressing issue.

Thanks for clarifying, all good.

**Comment by b1rger** [comment]
> This is not the first time that I see that several people had to re-discover features or implications of a tool that we use across projects;

it took me 20seconds to do a google search for "typesense search multiple collections" and find the official typesense documentation [federated search](https://typesense.org/docs/guide/reference-implementations/federated-search.html). so I'm not sure it's that big of an issue

> and I see a deeper issue there that we lack a place where we can reliably deposit such information or a structure or space where we can share such information regularly so that people actually can find it.

Feel free to document your findings in the https://github.com/acdh-oeaw/prosnet-team repository - I'd be very interested in your arguments against federated search
(to me, documenting the reasoning behind the decision in *this* issue seems more reasonable, but YMMV)
